src/praisonai-code/praisonai_code/cli/features/doctor/checks/env_checks.py

```
@@ -482,37 +506,77 @@ def check_optional_deps(config: DoctorConfig) -> CheckResult:
+            remaining = max(0.0, deadline - time.monotonic())
+            wait_for = min(per_package_timeout, remaining) if remaining > 0 else 0.0
+            try:
+                future.result(timeout=wait_for)
+                available.append(package)
+            except FuturesTimeoutError:
+                # Import is hanging or unusually slow; skip it rather than let
+                # it block the whole deep-doctor pass.
+                slow.append(f"{package} ({description})")
+            except ImportError:
+                missing.append(f"{package} ({description})")
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)